tools/generator/player.py

```diff
@@ -32,32 +67,50 @@ def gen_lang(lang: str, *, files, symlinks) -> None:
             else:
                 gen.template(name)
 
-    gen_lang('c',
-             files=['champion.c', 'Makefile'],
-             symlinks=['api.h', 'interface.cc', 'Makefile-c'])
-    gen_lang('caml',
-             files=['champion.ml', 'champion.mli', 'Makefile'],
-             symlinks=['api.ml', 'interface.cc', 'Makefile-caml'])
-    gen_lang('cs',
-             files=['champion.cs', 'api.cs', 'interface.cc', 'Makefile'],
-             symlinks=['Makefile-cs'])
-    gen_lang('cxx',
-             files=['champion.cc', 'Makefile'],
-             symlinks=['api.hh', 'interface.cc', 'Makefile-cxx'])
-    gen_lang('haskell',
-             files=['Champion.hs', 'Makefile'],
-             symlinks=['Api.hs', 'CApi.hsc', 'api.h', 'interface.cc',
-                       'Makefile-haskell'])
-    gen_lang('java',
-             files=['Champion.java', 'Makefile'],
-             symlinks=['Api.java', 'interface.cc', 'Makefile-java'])
-    gen_lang('php',
-             files=['champion.php', 'Makefile'],
-             symlinks=['api.php', 'interface.cc', 'Makefile-php'])
-    gen_lang('python',
-             files=['Champion.py', 'Makefile'],
-             symlinks=['api.py', 'interface.cc', 'Makefile-python'])
-    gen_lang('rust',
-             files=['Makefile', 'champion.rs'],
-             symlinks=['Cargo.toml', 'Makefile-rust', 'api.rs', 'ffi.rs',
-                       'api.h', 'interface.cc'])
+    for lang, files in LANGUAGES.items():
+        gen_lang(lang, **files)
+
+
+def check_compile(lang_dir: Path, lang: str, *, cleanup=True) -> bool:
+    """Try to `make` a player, returns `True` on success.
+    If `cleanup`, also run `make clean`."""
+
+    # Print on stderr
+    def eprint(*args, **kwargs):
+        print(*args, file=stderr, **kwargs)
+
+    success = True
+    eprint(f"{lang}... ", end='', flush=True)
+
+    try:
+        subprocess.run(
+            ['make'], universal_newlines=True, check=True,
+            capture_output=True, cwd=lang_dir,
+        )
+    except subprocess.CalledProcessError as e:
+        success = False
+        eprint("FAILED")
+        eprint(e.stderr)
+    else:
+        eprint("OK")
+
+    if cleanup:
```

Comment:
I don't think this is needed, if you call `--check` on an already existing dir it's probably because you want to debug with the target output files. Otherwise you should use a `checkplayer` command that generates everything in a temporary directory, and here it will all get wiped out when the command ends anyway.